### PR TITLE
MICRO-7: Updated registration flow

### PR DESCRIPTION
**Pull request by johnjcsmith**
This PR implements the updated, multi-step registration flow.

The initial Magento `registration` happens as part of the Serverless deployment with the `registration_token` and `service_id` written to SSM for the activator function.
Once the stack update has completed, the `interface_context` is written to SSM. This contains the APIGateway APIKey and base url.

The activator function runs every 5 minutes and attempts to "activate" the Magento integration, create the config interface in Magento and store the magento `access_token` in SSM for serverless functions. This will fail until the integration has been approved. If the `access_token` has already been set, this will be skipped.

At the moment the activator function will continue triggering every 5 minutes. This will be resolved in a future feature.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -66,49 +99,183 @@ class ServerlessMagento {
...
+               .map((resource: any) => resource.PhysicalResourceId)
+
+               if (apiRestApis.length > 1) {
+                    throw new Error('Multiple APIGatewayRestApis not supported');
+               } else if (apiRestApis.length == 0) {
+                    throw new Error('Multiple APIGatewayRestApis');
+               }
+
+               const restApiId = apiRestApis.pop();
+               return `https://${restApiId}.execute-api.ap-southeast-2.amazonaws.com/${this.serverless.service.provider.stage}/`
```
Comment (AdamJHall) [inline src/index.ts:268]: Need to swap this out to the region of the service instead of hardcoding southeast-2

File: src/lib/activator.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+               } else {
+                    console.log('Service already activated.');
+               }
+
+          }`;
+
+          await fs.mkdir(handlerFolder, { recursive: true });
+          exec('npm init -y', { cwd: handlerFolder });
+          exec('npm install --save aws-sdk', { cwd: handlerFolder });
```
Comment (AdamJHall) [inline src/lib/activator.ts:216]: If we make these dependencies of this module this wouldn't be needed right? 

Just not sure how I feel about executing npm like this

File: src/index.ts (hunk 1 above)
Comment (johnjcsmith) [reply src/index.ts:268]: Good spot! Thanks.

File: src/lib/activator.ts (hunk 2 above)
Comment (johnjcsmith) [reply src/lib/activator.ts:216]: No. These are deps of the activator handler not this module. So they get built into the temporary handler build folder.

Comment (johnjcsmith) [reply src/lib/activator.ts:216]: I believe we *could* add these as dependencies of the serverless application which is including this plugin however then we would have to advise anyone who uses the plugin that they will need these deps available.

Comment (johnjcsmith) [reply src/lib/activator.ts:216]: I.e the dependencies of `@aligent/serverless-magento` wont be included.

**Review by AdamJHall — APPROVED** [review]
Discussed my comments in person, and ran through your reasoning. We've added some issues we can look at going forward so happy to approve this now :)

**Pull request merged**